File: Algorithms/p198_House_Robber_2.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int s0 = 0; // no robbing
        int s1 = 0; // robbing
        for(int v : nums){
            int tmp = s0;
            s0 = max(s0,s1);
            s1 = tmp+v;
        }
        return max(s0,s1);
    }
};

File: Algorithms/p198_House_Robber/p198_House_Robber_2.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        s0 = 0 # no robbing
        s1 = 0 # robbing
        for v in nums:
            s0, s1 = max(s0,s1), s0+v
        return max(s0,s1)
